controllers/dockermachine_controller.go

```diff
@@ -198,16 +181,20 @@ func (r *DockerMachineReconciler) create(
 			return ctrl.Result{}, nil
 		}
 
-		// create a control plane
-		r.Log.Info("Creating a brand new cluster")
-		elb, err := getExternalLoadBalancerNode(c.Name, r.Log)
+		r.Log.Info("Creating a brand new controlplane node")
+		elb, err := actions.GetExternalLoadBalancerNode(c.Name)
```

Comment:
This seems to be getting into implementation details of the `DockerCluster`, instead this should probably just check `Cluster.Status.InfrastructureReady` for the cluster that owns this `DockerMachine`

Reply:
Relying on `cluster.Status.APIEndpoints` for now. I didn't see the `InfrastructureReady` flag being set anywhere. Wanted to get this PR done to unblock @chuckha ASAP :) Will clean up  further in a follow-up?

Reply:
I'm good with punting for now, but lets make sure to create a tracking ticket to clean up later.

Reply:
Yes,  I'll start on that right after this is done.